src/state/farms/elastic/hooks.ts

```diff
@@ -0,0 +1,769 @@
+              rewardTokens: pool.rewardTokens.map(({ token }) => {
+                return token.id === ZERO_ADDRESS
+                  ? nativeOnChain(chainId)
+                  : new Token(chainId, token.id, Number(token.decimals), token.symbol, token.name)
+              }),
+              totalRewards: pool.rewardTokens.map(({ token, amount }) => {
+                const t =
+                  token.id === ZERO_ADDRESS
+                    ? nativeOnChain(chainId)
+                    : new Token(chainId, token.id, Number(token.decimals), token.symbol, token.name)
+                return CurrencyAmount.fromRawAmount(t, amount)
+              }),
+              tvlToken0,
+              tvlToken1,
+            }
+          }),
+        }
+      })
+      dispatch(setFarms({ chainId, farms: formattedData }))
+    }
+  }, [data, dispatch, chainId])
+
+  const multicallContract = useMulticallContract()
+
+  const getUserFarmInfo = useCallback(async () => {
+    const farmAddresses = elasticFarm.farms?.map(farm => farm.id)
+
+    if (chainId && account && farmAddresses?.length && multicallContract) {
+      console.time('getUserFarmInfo')
+      // get userDepositedNFTs
+      const userDepositedNFTsFragment = farmInterface.getFunction('getDepositedNFTs')
+      const callData = farmInterface.encodeFunctionData(userDepositedNFTsFragment, [account])
+
+      const chunks = farmAddresses.map(address => ({
+        target: address,
+        callData,
+      }))
+
+      const multicallRes = await multicallContract.callStatic.tryBlockAndAggregate(false, chunks)
+      const returnData = multicallRes.returnData
+      // listNFTs by contract
+      const nftResults: Array<Array<BigNumber>> = returnData.map((data: [boolean, string]) =>
+        data[0] ? farmInterface.decodeFunctionResult(userDepositedNFTsFragment, data[1]).listNFTs : [],
+      )
+
+      /*
+       * GET DETAIL NFT
+       */
+      const allNFTs = nftResults.flat()
+      const nftDetailFragment = positionManagerInterface.getFunction('positions')
+      const nftDetailChunks = allNFTs.map(id => ({
+        target: NETWORKS_INFO[chainId].elastic.nonfungiblePositionManager,
+        callData: positionManagerInterface.encodeFunctionData(nftDetailFragment, [id]),
+      }))
+
+      const detailNFTMultiCallData = (await multicallContract.callStatic.tryBlockAndAggregate(false, nftDetailChunks))
+        .returnData
+
+      const nftDetailResult = detailNFTMultiCallData.map((data: [boolean, string]) =>
+        data[0] ? positionManagerInterface.decodeFunctionResult(nftDetailFragment, data[1]) : null,
+      )
+
+      type NFT_INFO = {
+        [id: string]: {
+          poolAddress: string
+          liquidity: BigNumber
+          tickLower: BigNumber
+          tickUpper: BigNumber
+        }
+      }
+      const nftInfos = nftDetailResult.reduce((acc: NFT_INFO, item: any, index: number) => {
+        if (!item) return acc
+        return {
+          ...acc,
```

Comment:
should avoid this since this has complexity O(n2)

Reply:
the number nft in farm of one user is not too much, this is very fast compared with time to load it from rpc. LOL

Reply:
Yea, for this case I just noted "should avoid" since we may coding like this in other `.reduce` cases and creating performance issue.